examples/dex.py

```
@@ -0,0 +1,90 @@
+    storage[msg['sender']] = 1000
+
+elif msg['data'] == 'buy':
+    # User sends native coins to buy DEX tokens
+    if storage.get('k') is None:
+        raise Exception("Not initialized")
+    if msg['value'] <= 0:
+        raise Exception("Must send coins to buy tokens")
+    
+    # Calculate how many tokens to give using x * y = k
+    # (native_reserve + msg['value']) * (token_reserve - tokens_out) = k
+    new_native_reserve = storage['native_reserve'] + msg['value']
```

Ruff (0.15.15)

[error] Undefined name `storage`

(F821)